File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode dummy = new ListNode(-1);
        int sum = 0;
        ListNode curr = dummy;

        while(head.next != null) {
            head = head.next;
            if(head.val == 0) {
                curr.next = new ListNode(sum);
                sum = 0;
                curr = curr.next;
            }
            else {
                sum += head.val;
            }
        }

        return dummy.next;
    }
}

File: 2299-merge-nodes-in-between-zeros/2299-merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeNodes(self, head):
        dummy = ListNode(-1)
        sum = 0
        curr = dummy

        while head.next:
            head = head.next
            if head.val == 0:
                curr.next = ListNode(sum)
                sum = 0
                curr = curr.next
            else:
                sum += head.val
        
        return dummy.next
        